connections/ethereum/connection_test.go

```diff
@@ -61,8 +62,12 @@ func TestContractCode(t *testing.T) {
 }
 
 func TestConnection_SafeEstimateGas(t *testing.T) {
+	// In the case of d := c.Add(a, b), since c==d, there is a risk that MaxGasPrice itself will be changed,
+	// so the local variable maxGasPrice is defined by big.NewInt(0).
+	maxGasPrice := big.NewInt(0)
 	// MaxGasPrice is the constant price on the dev network, so we increase it here by 1 to ensure it adjusts
-	conn := NewConnection(TestEndpoint, false, AliceKp, log15.Root(), GasLimit, MaxGasPrice.Add(MaxGasPrice, big.NewInt(1)), GasMultipler, "", "")
+	maxGasPrice.Add(MaxGasPrice, big.NewInt(1))
+	conn := NewConnection(TestEndpoint, false, AliceKp, log15.Root(), GasLimit, maxGasPrice, MinGasPrice, GasMultipler, "", "")
```

Comment:
I have modified the way BigInt is added to MaxGasPrice.
The intention of the modification is that `d = c.Add(a, b)` will result in `d == c`, which may change the global variable MaxGasPrice from [`big.NewInt(ethutils.DefaultMaxGasPrice)`](https://github.com/ChainSafe/ChainBridge/pull/689/files#diff-d16ed2d5a9121d43922a239ea51f5e97b89a0e04761905bca49fc7ad253c13ccR21).

Reply:
Good catch